src/Networks/Network.ts

```diff
@@ -0,0 +1,323 @@
+        }
+        if (!Validate.isUndefined(props.dhcpd_dns_enabled)) {
+            this.dhcpd_dns_enabled = props.dhcpd_dns_enabled;
+        }
+        if (!Validate.isUndefined(props.gateway_type)) {
+            this.gateway_type = props.gateway_type;
+        }
+        if (!Validate.isUndefined(props.dhcpd_relay_enabled)) {
+            this.dhcpd_relay_enabled = props.dhcpd_relay_enabled;
+        }
+        if (!Validate.isUndefined(props.dhcpd_boot_enabled)) {
+            this.dhcpd_boot_enabled = props.dhcpd_boot_enabled;
+        }
+        if (!Validate.isUndefined(props.igmp_proxy_downstream)) {
+            this.igmp_proxy_downstream = props.igmp_proxy_downstream;
+        }
+        if (!Validate.isUndefined(props.upnp_lan_enabled)) {
+            this.upnp_lan_enabled = props.upnp_lan_enabled;
+        }
+        if (!Validate.isUndefined(props.dhcpd_ntp_enabled)) {
+            this.dhcpd_ntp_enabled = props.dhcpd_ntp_enabled;
+        }
+        if (!Validate.isUndefined(props.mdns_enabled)) {
+            this.mdns_enabled = props.mdns_enabled;
+        }
+        if (!Validate.isUndefined(props.lte_lan_enabled)) {
+            this.lte_lan_enabled = props.lte_lan_enabled;
+        }
+        if (!Validate.isUndefined(props.dhpcd_tftp_server)) {
+            this.dhpcd_tftp_server = props.dhpcd_tftp_server;
+        }
+        if (!Validate.isUndefined(props.auto_scale_enabled)) {
+            this.auto_scale_enabled = props.auto_scale_enabled;
+        }
+        if (!Validate.isUndefined(props.vlan)) {
+            this.vlan = props.vlan;
+        }
+        if (!Validate.isUndefined(props.l2tp_allow_weak_ciphers)) {
+            this.l2tp_allow_weak_ciphers = props.l2tp_allow_weak_ciphers;
+        }
+        if (!Validate.isUndefined(props.l2tp_local_wan_ip)) {
+            this.l2tp_local_wan_ip = props.l2tp_local_wan_ip;
+        }
+        if (!Validate.isUndefined(props.vpn_type)) {
+            this.vpn_type = props.vpn_type;
+        }
+        if (!Validate.isUndefined(props.radiusprofile_id)) {
+            this.radiusprofile_id = props.radiusprofile_id;
+        }
+        if (!Validate.isUndefined(props.dhpcd_wins_enabled)) {
+            this.dhpcd_wins_enabled = props.dhpcd_wins_enabled;
+        }
+        if (!Validate.isUndefined(props.require_mschapv2)) {
+            this.require_mschapv2 = props.require_mschapv2;
+        }
+        if (!Validate.isUndefined(props.l2tp_interface)) {
+            this.l2tp_interface = props.l2tp_interface;
+        }
+        if (!Validate.isUndefined(props.exposed_to_site_vpn)) {
+            this.exposed_to_site_vpn = props.exposed_to_site_vpn;
+        }
+        if (!Validate.isUndefined(props.x_ipsec_pre_shared_key)) {
+            this.x_ipsec_pre_shared_key = props.x_ipsec_pre_shared_key;
+        }
+
+        return this;
+    }
+
+    public async update(props: Partial<INetwork>): Promise<void> {
+        if (!Validate.isDefinedNotNull(props._id)) {
```

Comment:
```suggestion
        if (!Validate.isDefinedNotNull(this._id)) {
```
not sure what you want to do here ?

Reply:
With update() you'd requested the ability to update with out querying.  So my thought was that update() would need to have the _id passed in as part of the properties to update.  However after reviewing you can't create a Network without an id and we're more interested in ensuring it's available on the Network it self consistently.

Good catch.

Reply:
@KyleBrandon if you want to do this, it's better to do this with a static method / or a method on "networkS" 

There is no "sense" to get a network, and use it to update another (I think) ?

Reply:
I'm good with the change.  My thinking was flawed.